pkg/shipper/shipper_test.go

```diff
@@ -93,21 +87,35 @@ func TestShipper_UploadBlocks(t *testing.T) {
 
 		testutil.Ok(t, enc.Encode(&meta))
 
+		expBlocks[id] = struct{}{}
+
 		expFiles[id.String()+"/meta.json"] = buf.Bytes()
 		expFiles[id.String()+"/index"] = []byte("indexcontents")
 		expFiles[id.String()+"/chunks/0001"] = []byte("chunkcontents1")
 		expFiles[id.String()+"/chunks/0002"] = []byte("chunkcontents2")
-	}
 
-	testutil.Equals(t, timestamp.FromTime(now), gotMinTime)
+		// The shipper meta file should show all blocks as uploaded.
+		shipMeta, err := ReadMetaFile(dir)
+		testutil.Ok(t, err)
+		testutil.Equals(t, &Meta{Version: 1, Uploaded: ids[:i+1]}, shipMeta)
+
+		// Verify timestamps were updated correctly.
+		minTotal, maxSync, err := shipper.Timestamps()
+		testutil.Ok(t, err)
+		testutil.Equals(t, timestamp.FromTime(now), minTotal)
+		testutil.Equals(t, meta.MaxTime, maxSync)
+	}
 
 	for id := range expBlocks {
 		ok, _ := bucket.Exists(nil, id.String())
 		testutil.Assert(t, ok, "block %s was not uploaded", id)
 	}
 	for fn, exp := range expFiles {
```

Comment:
Maybe worth to check also if nothing is uploaded unnecessarily?

Reply:
We would literally have to make up random data for that to happen :P

Reply:
hm.. yeah, if you reupload a block.. it will be still the same ;p